.github/workflows/build_and_push.yml

```
@@ -0,0 +1,25 @@
+name: build and push docker images
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - '*'
+env:
+  BRANCH_NAME: ${{ github.head_ref || github.ref_name }}
+jobs:
+  build-and-push:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Login to GHCR (GitHub Packages)
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Build docker image
+        run: ./docker/build --with-dev # TODO: remove --with_dev before merging
+      - name: Push SHA docker image
```

```suggestion
      - name: Push docker image
```
This pushes both branch and tag doesn't it? So this name is a bit misleading.

In some projects we delay pushing the branch tag until tests have passed - can't remember what we were doing in this one on buildkite.